pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1939,22 +1940,35 @@ public TopicStatsImpl getStats(boolean getPreciseBacklog, boolean subscriptionBa
                         ledgers.forEach(ledgerId -> {
                             CompletableFuture<Void> completableFuture = new CompletableFuture<>();
                             getLedgerMetadataFutures.add(completableFuture);
-                            brokerService.getPulsar().getBookKeeperClient()
-                                    .getLedgerMetadata(ledgerId)
-                                    .thenAccept(metadata -> {
-                                        LedgerInfo schemaLedgerInfo = new LedgerInfo();
-                                        schemaLedgerInfo.ledgerId = metadata.getLedgerId();
-                                        schemaLedgerInfo.entries = metadata.getLastEntryId() + 1;
-                                        schemaLedgerInfo.size = metadata.getLength();
-                                        if (includeLedgerMetadata) {
-                                            info.metadata = metadata.toSafeString();
-                                        }
-                                        stats.schemaLedgers.add(schemaLedgerInfo);
-                                        completableFuture.complete(null);
-                                    }).exceptionally(e -> {
-                                completableFuture.completeExceptionally(e);
-                                return null;
-                            });
+                            CompletableFuture<LedgerMetadata> metadataFuture =
+                                    null;
+                            try {
+                                metadataFuture = brokerService.getPulsar().getBookKeeperClient()
+                                .getLedgerMetadata(ledgerId);
+                            } catch (Exception e) {
```

Comment:
catching "Exception" is too broad here, it can be a network error for instance or an OutOfMemoryError

also getLedgerMetadata is a method that returns a CompletableFuture, so I do not expect it to throw Exceptions but to return failed futures

Reply:
which error are you seeing ?

Reply:
NullPointException, I don't want to fail the complete `get internal stats` request even if we can't read the schema ledger metadata, so print the log in the broker side.

Reply:
how can we see a NPE here ?
can we perform a null check on 
`brokerService.getPulsar().getBookKeeperClient()   `                            
or is it a bug inside `getLedgerMetadata` ?